apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -503,63 +508,60 @@ export function getLevelToolboxBlocks(customCategory: string) {
 export function getSimplifiedStateForFlyout(
   serialization: WorkspaceSerialization
 ) {
-  const variableMap: {[key: string]: string} = {};
-  const variables = serialization.variables;
+  const blocksList = [] as object[];
+
+  const {variables, blocks} = serialization;
+
+  // Create a map of variable ids and names from the serialization.
+  const serializedVariableMap: Map<string, string> = new Map();
   variables?.forEach(variable => {
-    variableMap[variable.id] = variable.name;
+    serializedVariableMap.set(variable.id, variable.name);
   });
 
-  const blocksList = hasBlocks(serialization)
-    ? serialization.blocks.blocks.map(block =>
-        simplifyBlockState(block, variableMap)
-      )
-    : [];
+  // Replace variable ids with names and simplify state for flyout.
+  blocks.blocks.forEach(block => {
+    updateVariableFields(block, serializedVariableMap);
+    blocksList.push(simplifyBlockState(block));
+  });
 
   return blocksList;
 }
 
+// Function for updating field values
+function updateVariableFields(
+  block: {fields: SerializedFields},
+  serializedVariableMap: Map<string, string>
+): void {
+  const fields = block.fields;
+  for (const key in fields) {
+    const field = fields[key];
+    if (field.id && serializedVariableMap.has(field.id)) {
+      field.name = serializedVariableMap.get(field.id)!;
+      delete field.id;
```

Comment:
Do we need to delete them? Can you think of any scenarios in which we'd need this on a subsequent run or something?